tests/test_cgan.py

```diff
@@ -120,29 +119,15 @@ def build_discriminator(self):
         return Model([img, label], validity)
 
 
-@unittest.skipIf(keras2onnx.proto.tfcompat.is_tf2 and is_tf_keras, "Tensorflow 1.x only tests.")
-class TestCGAN(unittest.TestCase):
-
-    def setUp(self):
-        self.model_files = []
-
-    def tearDown(self):
-        for fl in self.model_files:
-            os.remove(fl)
-
-    @unittest.skipIf(is_tf_keras and StrictVersion(tf.__version__) < StrictVersion("1.14.0"),
-                     "Not supported before tensorflow 1.14.0 for tf_keras")
-    def test_CGAN(self):
-        keras_model = CGAN().combined
-        batch = 5
-        x = np.random.rand(batch, 100).astype(np.float32)
-        y = np.random.rand(batch, 1).astype(np.float32)
-        expected = keras_model.predict([x, y])
-        onnx_model = keras2onnx.convert_keras(keras_model, keras_model.name)
-        self.assertTrue(run_onnx_runtime(onnx_model.graph.name, onnx_model,
-                                         {keras_model.input_names[0]: x, keras_model.input_names[1]: y}, expected,
-                                         self.model_files))
-
-
-if __name__ == "__main__":
-    unittest.main()
+@pytest.mark.skipif(keras2onnx.proto.tfcompat.is_tf2 and is_tf_keras, reason="Tensorflow 1.x only tests.")
+@pytest.mark.skipif(is_tf_keras and StrictVersion(tf.__version__) < StrictVersion("1.14.0"),
```

Comment:
StrictVersion(tensorflow.__version__.split('-')[0]) >= StrictVersi...
otherwise, the whole test cases will failed on tensorflow nightly build.

Reply:
I'm confused by what you mean here. The current conditional is the same as that in line 133, so the previous behavior is reproduced. Was the original incorrect?

Reply:
The previous code is also ‘imprecise’, but the issue was hidden because this condition wasn't tested due to the first is_tf2 test is already wrong. And it looks the pytest.mark will test both. If you install a tensorflow nightly build. you will see the failure.

Reply:
Ok, that makes sense. I've made the change.